cmd/kubeadm/app/preflight/checks.go

```diff
@@ -869,6 +870,25 @@ func (ncc NumCPUCheck) Check() (warnings, errorList []error) {
 	return warnings, errorList
 }
 
+// MemCheck checks if the number of megabytes of memory is not less than required
+type MemCheck struct {
+	Mem uint64
+}
+
+// Name returns the label for memory
+func (MemCheck) Name() string {
+	return "Mem"
+}
+
+// Check number of memory required by kubeadm
+func (mc MemCheck) Check() (warnings, errorList []error) {
+	actual := memory.TotalMemory() / 1024 / 1024 // TotalMemory returns bytes; convert to MB
+	if actual < mc.Mem {
+		errorList = append(errorList, errors.Errorf("the system RAM (%d MB) is less than the minimum %d MB", actual, mc.Mem))
```

Comment:
maybe we should treat it as a warning.

Reply:
@neolit123 I think too small memory may affect the stability of the cluster... 

and the cpu check will bring an error, I think the memory check should have the same behavior of cpu check.

Reply:
/cc @rosti

Reply:
I would suggest to keep it as an error, but lower the error threshold a bit. Often times folks might deploy control planes in machines that have a bit less than 2GB RAM (be it due to sharing RAM between the CPU & a built in GPU, or some other system reservation). Therefore it might be better to make it ~1700 MB. Below that amount of RAM running a stable control plane would be difficult.

Reply:
probably best to say "not recommended to have less than X memory"

Reply:
@neolit123 @rosti the CPU checker is an error, I think maybe we should keep the memory checker having the same behavior.

if the CPU checker and memory checker have different behavior, it will be a little confusing...